packages/next/src/server/app-render/app-render.tsx

```diff
@@ -2402,201 +2450,219 @@ async function spawnDynamicValidationInDev(
+    await cacheSignal.cacheReady()
+    initialClientController.abort()
+  }
 
   const finalServerController = new AbortController()
-  const serverDynamicTracking = createDynamicTrackingState(false)
+  const serverDynamicTracking = createDynamicTrackingState(
+    false // isDebugDynamicAccesses
+  )
 
-  const finalServerPrerenderStore: PrerenderStore = {
+  const finalRenderPrerenderStore: PrerenderStore = {
     type: 'prerender',
     phase: 'render',
     rootParams,
     implicitTags,
     renderSignal: finalServerController.signal,
     controller: finalServerController,
-    // During the final prerender we don't need to track cache access so we omit the signal
+    // All caches we could read must already be filled so no tracking is necessary
     cacheSignal: null,
     dynamicTracking: serverDynamicTracking,
     revalidate: INFINITE_CACHE,
     expire: INFINITE_CACHE,
     stale: INFINITE_CACHE,
-    tags: [],
+    tags: [...implicitTags.tags],
     prerenderResumeDataCache,
     hmrRefreshHash,
   }
 
-  const finalClientController = new AbortController()
-  const clientDynamicTracking = createDynamicTrackingState(false)
-  const dynamicValidation = createDynamicValidationState()
+  const finalAttemptRSCPayload = await workUnitAsyncStorage.run(
+    finalRenderPrerenderStore,
+    getRSCPayload,
+    tree,
+    ctx,
+    isNotFound
+  )
+  const reactServerResult = await createReactServerPrerenderResult(
+    prerenderAndAbortInSequentialTasks(
+      async () => {
+        const prerenderResult = await workUnitAsyncStorage.run(
+          // The store to scope
+          finalRenderPrerenderStore,
+          // The function to run
+          ComponentMod.prerender,
+          // ... the arguments for the function to run
+          finalAttemptRSCPayload,
+          clientReferenceManifest.clientModules,
+          {
+            onError: (err: unknown) => {
+              if (
+                finalServerController.signal.aborted &&
+                isPrerenderInterruptedError(err)
+              ) {
+                return err.digest
+              }
 
+              return getDigestForWellKnownError(err)
+            },
+            signal: finalServerController.signal,
+          }
+        )
+        return prerenderResult
+      },
+      () => {
+        finalServerController.abort()
+      }
+    )
+  )
+
+  const clientDynamicTracking = createDynamicTrackingState(
+    false //isDebugDynamicAccesses
+  )
+  const finalClientController = new AbortController()
   const finalClientPrerenderStore: PrerenderStore = {
     type: 'prerender',
     phase: 'render',
     rootParams,
     implicitTags,
     renderSignal: finalClientController.signal,
     controller: finalClientController,
-    // During the final prerender we don't need to track cache access so we omit the signal
+    // For HTML Generation the only cache tracked activity
+    // is module loading, which has it's own cache signal
```

Comment:
Nit: Module loading is not tracked during the _final_ client prerender though. I think the previous comment was correct here.